string desired_string
---
string new_string